src/batList.mlv

```diff
@@ -1304,6 +1304,15 @@ let subset cmp l l' = for_all (fun x -> mem_cmp cmp x l') l
   subset Pervasives.compare [1;2] [1;2;3] = true
 *)
 
+let shuffle ?state l =
+  let a = BatInnerShuffle.array_shuffle ?state (Array.of_list l) in
+  Array.to_list a
+(*$T shuffle
+  let s = Random.State.make [|11|] in \
+  shuffle ~state:s [1;2;3;4;5;6;7;8;9] = [7; 2; 9; 5; 3; 6; 4; 1; 8]
+  shuffle [] = []
+*)
```

Comment:
since I think the remark above is actually a bug in the old implementation, maybe a random test that checks whether the new array is a permutation of the old one would be helpful.
Something like:

```ocaml
(*$Q
  Q.(array_of_size Gen.(2--15) small_int) (fun a -> \
    let a' = Array.copy a in \
    shuffle a'; \
    (Array.to_list a' |> List.sort Pervasives.compare) = \
     (Array.to_list a |> List.sort Pervasives.compare))
*)
```

`(*$Q
  Q.(small_list small_int) (fun l ->\